src/llm_vm/client.py

```diff
@@ -109,7 +127,12 @@ def complete(self, prompt,
         static_context = context
         dynamic_prompt = prompt
         use_rebel_agent = False
-
+        # TODO(us): speculative_sampling gives error when temp is 0,
+        # self.optimizer is None means `speculative_sampling` is used
```

Comment:
Shouldn’t we add an explicit flag for speculation?

Reply:
in the current code, as default mode, speculative sampling and as optional normal optimizer can be used, as default should I keep the current optimizer?